app/models/project.rb

```diff
@@ -1,5 +1,19 @@
 # frozen_string_literal: true
 class Project < ActiveRecord::Base
+  # Used as part of temporary abstract attribute for docker build radio buttons
+  DOCKER_BUILD_METHODS = [
+    {
+      label: 'Docker images build externally',
+      method: 'docker_image_building_disabled',
+      help_text: 'Disable local building of docker images, they must be added via api.'
+    },
+    {
+      label: 'Samson manages docker images',
+      method: 'samson',
+      help_text: ''
+    }
+  ] + Samson::Hooks.fire(:project_docker_build_method_options).flatten
```

Comment:
`.flatten` is usually a bad idea since it's expensive ... `.flatten(1)` is what you want here

`[[[1]]].flatten(1) -> [[1]]`